### PR TITLE
test: Fix hot restart test under --runs_per_test

### PR DESCRIPTION
**Pull request by dnoe**
*Description*:
When using --runs_per_test with concurrency the individual invocations
of the hot restart test may collide. Bazel provides an environment
variable when using --runs_per_test that allows us to set --base_id in
each parallel invocation to a guaranteed unique value so this doesn't
happen.

*Risk Level*: Low

*Testing*: bazel test //test/integration:hotrestart_test --runs_per_test=100

Signed-off-by: Dan Noé <dpn@google.com>

### CODE REVIEW

File: test/integration/hotrestart_test.sh [hunk 1]
```diff
@@ -45,20 +46,29 @@ if [[ -z "${ENVOY_IP_TEST_VERSIONS}" ]] || [[ "${ENVOY_IP_TEST_VERSIONS}" == "al
   JSON_TEST_ARRAY+=("${HOT_RESTART_JSON_V6}")
 fi
 
+# Enable this test to work with --runs_per_test
+if [[ -z "${TEST_RANDOM_SEED}" ]]; then
+  BASE_ID=1
+else
+  BASE_ID="${TEST_RANDOM_SEED}"
```
Comment (htuch) [inline test/integration/hotrestart_test.sh]: Arguably this still has the issue that if you run two distinct `bazel test` on the same machine, with different run IDs, you will have potential conflict. I think it's fine for now, this is a useful improvement over status quo.

Comment (dnoe) [reply test/integration/hotrestart_test.sh]: Doesn't Bazel serialize separate build or test invocations made on the same machine in the same code base? When I try it in a separate window it tells me it's waiting for the first command to complete.

Comment (ggreenway) [reply test/integration/hotrestart_test.sh]: I think it's per-bazel-daemon.  On an interactive system, that's generally 1 per user (I think).  On a build machine, it's probably 1 per container.

**Comment by dnoe** [comment]
Amusingly, looks like it has failed in CI on the real flake in hotrestart test in async grpc client

**Comment by dnoe** [comment]
I think all the cases where the envoy binary is launched from the hot restart test are now properly configured with base-id.

File: test/integration/hotrestart_test.sh (hunk 1 above)
Comment (htuch) [reply test/integration/hotrestart_test.sh]: It's per source directory actually (I know because I run multiple concurrently in different Envoy trees :).

**Comment by htuch** [comment]
Please merge master to pick up https://github.com/envoyproxy/envoy/pull/2613, this will unblock CI.

**Pull request merged**